include/geode/model/representation/builder/detail/copy.h

```diff
@@ -117,6 +211,37 @@ namespace geode
             return mapping;
         }
 
+        template < typename ModelFrom, typename BuilderTo >
+        void copy_block_components(
+            const ModelFrom& from, BuilderTo& builder_to, Mapping& mapping )
+        {
+            for( const auto& block : from.blocks() )
+            {
+                if( mapping.has_mapping_input( block.id() ) )
+                {
+                    const auto& id = mapping.in2out( block.id() );
+                    if( ModelFrom::dim == BuilderTo::dim )
+                    {
+                        builder_to.add_block( id, block.mesh().impl_name() );
+                    }
+                    else
+                    {
+                        builder_to.add_block( id );
+                    }
+                    builder_to.set_block_name( id, block.name() );
+                }
+                else
+                {
+                    const auto& id =
+                        ModelFrom::dim == BuilderTo::dim
+                            ? builder_to.add_block( block.mesh().impl_name() )
+                            : builder_to.add_block();
```

Comment:
is impl_name dependant of the dimension?

Reply:
Yes, impl_name will return the MeshImpl, which is dependant on the dimension (ex: OpenGeodePointSet2D)